src/util/flattenByGroupAllClass.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Metric,
+  NullSketchCollection,
+  SketchCollection,
+  firstMatchingMetric,
+  groupBy,
+  keyBy,
+} from "@seasketch/geoprocessing/client-core";
+
+/**
+ * Returns one aggregate object for every groupId present in metrics
+ * Each object includes following properties:
+ * numSketches - count of child sketches in the group
+ * [classId] - a percValue for each classId present in metrics for group
+ * value - sum of value across all classIds present in metrics for group
+ * percValue - given sum value across all classIds, contains ratio of total sum across all class IDs
+ */
+export const flattenByGroupAllClass = (
+  collection: SketchCollection | NullSketchCollection,
+  /** Group metrics for collection and its child sketches */
+  groupMetrics: Metric[],
+  /** Totals by class */
+  totalMetrics: Metric[]
+): {
+  value: number;
+  groupId: string;
+  percValue: number;
+}[] => {
```

Comment:
Is there a reason the version of this in gp can't be used directly?  Code comment addition, explaining this would be helpful. especially when comes time to migrate back into gp lib.

Reply:
Added a code comment above describing the differences
https://github.com/seasketch/azores-nearshore-reports/blob/692a09dab82cfa76ea3c6956c32cbe74d3c5b25d/src/util/flattenByGroupAllClass.ts#L17-L19